### DISCUSSION TITLE
What are the time units used by the artillery-engine-grpc reports

### DISCUSSION THREAD

**Discussion by fage88** [artilleryio/artillery#1186] [Artillery]
I have an API I have implemented in both REST and gRPC.
I am trying to load test them to prove which is more performant.
I am using artillery and the grpc plugin to accomplish this.
The output I get for my gRPC load test is 
```All virtual users finished
Summary report @ 13:40:48(-0400) 2021-10-18
  Scenarios launched:  300
  Scenarios completed: 296
  Requests completed:  0
  Mean response/sec: NaN
  Response time (msec):
    min: NaN
    max: NaN
    median: NaN
    p95: NaN
    p99: NaN
  Scenario counts:
    create and view: 300 (100%)
  Errors:
    Error: 6 ALREADY_EXISTS: A book with ISBN 43219658 already exists: 1
    Error: 6 ALREADY_EXISTS: A publisher with code Wljz already exists: 1
    Error: 6 ALREADY_EXISTS: A book with ISBN 21721103 already exists: 1
    Error: 6 ALREADY_EXISTS: A publisher with code jmn already exists: 1
  engine.grpc.response_time:
    min: 112.7
    max: 299631.8
    median: 151242
    p95: 285105.7
    p99: 297016.3
  engine.grpc.responses.total: 4756
  engine.grpc.responses.success: 4752
  engine.grpc.codes.0: 4752
  engine.grpc.responses.error: 4
  engine.grpc.codes.6: 4
```
But I am not 100% sure what time unit the `engine.grpc.response_time` value is in.  My guess is it is in nano seconds but I am not sure.

Can anyone let me know what this value is in or if there is a way in my config where I define the grpc engine can I set the time units?

**Comment by hassy** [comment]
@fage88 it looks like it's milliseconds: https://github.com/kenju/artillery-engine-grpc/blob/master/index.js#L134

**Reply by fage88** [reply]
Thank you @hassy
